test/unit-test/stun_deserializer/stun_deserializer_utest.c

```diff
@@ -1649,3 +1649,429 @@ void test_StunDeserializer_ParseAttributeAddress_XorPeer_HappyPath( void )
+
+    attribute.attributeType = STUN_ATTRIBUTE_TYPE_XOR_MAPPED_ADDRESS;
+    attribute.pAttributeValue = &( attributeValue[0] );
+    attribute.attributeValueLength = sizeof( attributeValue );
+
+    result = StunDeserializer_ParseAttributeAddress( &( ctx ),
+                                                     &( attribute ),
+                                                     &( address ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
+    TEST_ASSERT_EQUAL( STUN_ADDRESS_IPv6,
+                       address.family );
+    TEST_ASSERT_EQUAL( ( 0x1234 ^ ( STUN_HEADER_MAGIC_COOKIE >> 16 ) ),
+                       address.port );
+    TEST_ASSERT_EQUAL_UINT8_ARRAY( expectedAddress,
+                                   address.address,
+                                   16 );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate StunDeserializer_FindAttribute in case of different attribute types.
+ *
+ * It includes the following scenarios:
+ *
+ * 1. Passing a NULL pointer for the StunAttribute_t structure, which should return
+ *    STUN_RESULT_BAD_PARAM.
+ * 2. Trying to find the STUN_ATTRIBUTE_TYPE_ICE_CONTROLLED attribute, which is not
+ *    present in the STUN message, and expecting the function to return
+ *    STUN_RESULT_NO_ATTRIBUTE_FOUND.
+ * 3. Finding the STUN_ATTRIBUTE_TYPE_PRIORITY attribute, which is present in the
+ *    STUN message, and verifying that the function returns STUN_RESULT_OK.
+ */
+void test_StunDeserializer_FindAttribute( void )
+{
+    StunContext_t ctx = { 0 };
+    StunResult_t result;
+    StunHeader_t header = { 0 };
+    StunAttribute_t attribute = { 0 };
+    uint8_t serializedMessage[] =
+    {
+        /* Message Type = STUN Binding Request, Message Length = 0x08 (excluding 20 bytes header). */
+        0x00, 0x01, 0x00, 0x08,
+        /* Magic cookie. */
+        0x21, 0x12, 0xA4, 0x42,
+        /* Transaction ID. */
+        0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC, 0xDE, 0xF0, 0xAB, 0xCD, 0xEF, 0xA5,
+        /* Attribute Type = PRIORITY (0x0024), Attribute Length = 4. */
+        0x00, 0x24, 0x00, 0x04,
+        /* Attribute Value: 0x6E000100 (Priority = 2023406816). */
+        0x6E, 0x00, 0x01, 0x00,
+    };
+    size_t serializedMessageLength = sizeof( serializedMessage );
+    uint8_t attributeValue[] = {0x6E, 0x00, 0x01, 0x00,};
+    result = StunDeserializer_Init( &( ctx ),
+                                    &( serializedMessage[ 0 ] ),
+                                    serializedMessageLength,
+                                    &( header ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
+
+    result = StunDeserializer_FindAttribute( &( ctx ),
+                                             STUN_ATTRIBUTE_TYPE_ICE_CONTROLLED,
+                                             NULL );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_BAD_PARAM,
+                       result );
```

Comment:
I'd suggest to move this one to next test case `test_StunDeserializer_FindAttribute_BadParams` for better categorization.

Reply:
Yes, I have now added separate test cases.